### PR TITLE
Act tree/new tree

### PR DESCRIPTION
**Pull request by Neill83**

### CODE REVIEW

File: src/app/tree/node-tree/data/terra-node-tree.config.ts [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+            console.error('Node ' + node.name + ' with id ' + node.id + ' not found!');
+        }
+        else
+        {
+            this.internalRemoveNode(node);
+        }
+    }
+
+    //removes a given node
+    private internalRemoveNode(node:TerraNodeInterface<D>)
```
Comment (MFrank84) [inline src/app/tree/node-tree/data/terra-node-tree.config.ts]: return type

File: src/app/tree/node-tree/data/terra-node-tree.config.ts [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+        {
+            this._currentSelectedNode = null;
+        }
+    }
+
+    /**
+     * @description Removes a node by ID.
+     * @param id The ID of the node to be removed.
+     */
+    public removeNodeById(id:string | number)
```
Comment (MFrank84) [inline src/app/tree/node-tree/data/terra-node-tree.config.ts]: return type

File: src/app/tree/node-tree/data/terra-node-tree.config.ts [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+                this.recursiveSetNodeInactive(node.children);
+            }
+        });
+    }
+
+    public get searchNodeList():Array<TerraSuggestionBoxValueInterface>
+    {
+        return this._searchNodeList;
+    }
+}
```
Comment (MFrank84) [inline src/app/tree/node-tree/data/terra-node-tree.config.ts]: newline

File: src/app/tree/node-tree/data/terra-node.interface.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * @description Lazy loading function to get data from server. Optional.
+     */
+    onLazyLoad?:()=>Observable<any>;
+
+    /**
+     * @description Check if lazy loading has finished to avoid firing a REST-Call again
+     */
+    hasLoaded?:boolean;
+}
```
Comment (MFrank84) [inline src/app/tree/node-tree/data/terra-node.interface.ts]: newline

File: src/app/tree/node-tree/node/terra-node.component.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    this.inputNode.hasLoaded = false;
+                    this.isLoading = false;
+                });
+        }
+        else
+        {
+            this.inputNode.isOpen = !this.inputNode.isOpen;
+        }
+    }
+}
```
Comment (MFrank84) [inline src/app/tree/node-tree/node/terra-node.component.ts]: newline

File: src/app/tree/node-tree/terra-node-tree.component.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<ng-content></ng-content>
+<ul class="terra-node-tree">
+    <ng-template ngFor let-node [ngForOf]="inputConfig.list">
+        <terra-node [inputNode]="node" [hidden]="node.isHidden" [inputConfig]="inputConfig"></terra-node>
+    </ng-template>
+</ul>
```
Comment (MFrank84) [inline src/app/tree/node-tree/terra-node-tree.component.html]: newline

File: src/app/tree/node-tree/terra-node-tree.component.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<ng-content></ng-content>
+<ul class="terra-node-tree">
+    <ng-template ngFor let-node [ngForOf]="inputConfig.list">
```
Comment (MFrank84) [inline src/app/tree/node-tree/terra-node-tree.component.html]: ng-container ist die bessere variante zu ng-template 
https://angular.io/guide/structural-directives#ngcontainer

File: src/app/tree/node-tree/terra-node-tree.component.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    public ngOnInit():void
+    {
+    }
+
+    public ngOnDestroy():void
+    {
+        this.inputConfig.reset();
+    }
+}
```
Comment (MFrank84) [inline src/app/tree/node-tree/terra-node-tree.component.ts]: newline

File: src/app/tree/node-tree/terra-node-tree.component.ts [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+            this.recursiveCheckList(this.inputConfig.list);
+        }
+        else
+        {
+            this.inputConfig.toggleVisiblityForAllChildren(this.inputConfig.list, true);
+            this.inputConfig.closeAllNodes();
+        }
+    }
+
+    private recursiveCheckList(list:Array<TerraNodeInterface<D>>)
```
Comment (garvins) [inline src/app/tree/node-tree/terra-node-tree.component.ts]: return type

File: src/app/tree/node-tree/terra-node-tree.component.ts [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        let suggestion:string = name.toUpperCase();
+
+        // check if search string is included in the given suggestion
+        if(suggestion.includes(value.toUpperCase()))
+        {
+            this.handleNodeVisibility(node);
+        }
+    }
+
+    private handleNodeVisibility(node:TerraNodeInterface<D>)
```
Comment (garvins) [inline src/app/tree/node-tree/terra-node-tree.component.ts]: return type

**Pull request merged**